users/views.py

```diff
@@ -1,3 +1,112 @@
-from django.shortcuts import render
+import json, re, bcrypt, jwt, requests
 
-# Create your views here.
+from django.views         import View
+from django.http          import JsonResponse
+
+from users.models         import User
+
+from my_settings          import SECRET_KEY, const_algorithm, temporary_password, temporary_phone_number, rest_apt_key
+
+class SignUpView(View):
+    def post(self, request):
+        try:
+            data            = json.loads(request.body)
+            email           = data['email']
+            password        = data['password']
+            hashed_password = bcrypt.hashpw(password.encode('UTF-8'), bcrypt.gensalt())
+
+            if User.objects.filter(email=email).exists():
+                return JsonResponse({"MESSAGE": "EMAIL_ALREADY_EXIST"}, status=400)
+
+            if not re.match(r"^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$", email):
+                return JsonResponse({"MESSAGE": "INVALID_FORMAT"}, status=400)
+
+            if not re.match(r"^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$", password):
+                return JsonResponse({"MESSAGE": "INVALID_FORMAT"}, status=400)
+
+            User.objects.create(
+                email        =   email,
+                password     =   hashed_password.decode('UTF-8'),
+                phone_number =   data['phone_number'],
+                nickname     =   data.get('nickname'),
+                address      =   data.get('address'),
+                user_image   =   data.get('user_image'),
+                birthday     =   data.get('birthday'),
+            )
+            return JsonResponse({"MESSAGE": "SUCCESS"}, status=201)
+
+        except KeyError:
+            return JsonResponse({"MESSAGE": "KEY_ERROR"}, status=400)
+
+class SignInView(View):
+    def post(self, request):
+        try:
+            data     = json.loads(request.body)      
+            email    = data['email']
+            password = data['password']        
+
+            if not User.objects.filter(email = email).exists():
+                return JsonResponse({'MESSAGE':'INVALID_VALUE'}, status = 401)
+
+            if bcrypt.checkpw(password.encode('utf-8'),User.objects.get(email=email).password.encode('utf-8')):
+                token = jwt.encode({'id':User.objects.get(email=email).id}, SECRET_KEY, algorithm=const_algorithm)
```

Comment:
```suggestion
            user = User.objects.get(email=email)
            
            if bcrypt.checkpw(password.encode('utf-8'), user.password.encode('utf-8')):
                token = jwt.encode({'id' : user.id}, SECRET_KEY, algorithm=const_algorithm)
```
user를 미리 get을 한 뒤에 사용하면 각독성이 훨씬 올라가고, 중복으로 데이터베이스에 쿼리를 요청하지 않습니다.

Reply:
넵 ㅎㅎ